ocamldoc/odoc_sig.ml

```diff
@@ -282,23 +282,26 @@ module Analyser =
           | Ptyp_object (fields, _) ->
             let rec f = function
               | [] -> []
-              | ({txt=""},_,_) :: _ ->
+              | Otag ({txt=""},_,_) :: _ ->
                 (* Fields with no name have been eliminated previously. *)
                 assert false
-
-              | ({txt=name}, _atts, ct) :: [] ->
+              | Otag ({txt=name}, _atts, ct) :: [] ->
                 let pos = Loc.ptyp_end ct in
                 let (_,comment_opt) = just_after_special pos pos_end in
                 [name, comment_opt]
-              | ({txt=name}, _atts, ct) :: ((_name2, _atts2, ct2) as ele2) :: q ->
+              | Otag ({txt=name}, _, ct) ::
+                  ((Oinherit ct2) as ele2) :: q
+              | Otag ({txt=name}, _, ct) ::
+                  ((Otag (_, _, ct2)) as ele2) :: q ->
                 let pos = Loc.ptyp_end ct in
                 let pos2 = Loc.ptyp_start ct2 in
                 let (_,comment_opt) = just_after_special pos pos2 in
                 (name, comment_opt) :: (f (ele2 :: q))
+              | _ :: q -> f q
```

Comment:
@Octachron Thanks for your review. Coming back to this comment after three months, I am still not sure what to do here because inherited field gets expanded in documentation (in elsewhere as well, e.g. toplevel display). This means when we encounter an inherited field (`Oinherit of core_type`), we want to fetch documentation associated with it somewhere, which means we need to save doc generation result for all object type declaration. I am not sure how to handle cross-module/file case either. Any suggestion is welcomed here...